### PR TITLE
Add feature comparison table

### PR DESCRIPTION
**Pull request by navarone-feekery**
- Add a table with feature comparisons between Open Crawler and Elastic Crawler

(The table is a bit crazy in plaintext because of links, I recommend viewing the file directly)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -190,3 +190,33 @@ When given a command, Open Crawler will run until the process is finished.
...
+
+### Feature Comparison
+
+Elastic also has the cloud-base [Elastic Crawler](https://www.elastic.co/guide/en/enterprise-search/current/crawler.html) with similar features.
+Below is a comparison table below for the features that are available in Open Crawler vs Elastic Crawler (cloud).
+
+|                                                                                                                                                        | Open Crawler | Elastic Crawler (cloud) |
+|--------------------------------------------------------------------------------------------------------------------------------------------------------|--------------|-------------------------|
+| **Interface**                                                                                                                                          | CLI          | GUI (Kibana)            |
+| **Hosting**                                                                                                                                            | Self-hosted  | Elastic Cloud           |
```
Comment (seanstory) [inline README.md:202]:
```suggestion
| **Hosting**                                                                                                                                            | Self-hosted  | Elastic Cloud or Self-hosted Enterprise Search          |
```

File: README.md [hunk 2]
```diff
@@ -190,3 +190,33 @@ When given a command, Open Crawler will run until the process is finished.
...
+| **Indexing using `_bulk` API**                                                                                                                         | Yes          | No                      |
+| **[Ingest pipelines](./docs/features/INGEST_PIPELINES.md)**                                                                                            | Yes          | Yes                     |
+| **[Binary content extraction](./docs/features/BINARY_CONTENT_EXTRACTION.md)**                                                                          | Yes          | Yes                     |
+| **[Crawler directives](./docs/features/CRAWLER_DIRECTIVES.md)**                                                                                        | Yes *        | Yes                     |
+| **[Scheduling](#scheduling-recurring-crawl-jobs)**                                                                                                     | Yes          | Yes                     |
+| **Crawler results history and metadata**                                                                                                               | No **        | Yes                     |
```
Comment (seanstory) [inline README.md:211]:
```suggestion
| **Crawl result history and metadata**                                                                                                               | No **        | Yes                     |
```

File: README.md [hunk 3]
```diff
@@ -190,3 +190,33 @@ When given a command, Open Crawler will run until the process is finished.
...
+| **[Duplicate content handling](https://www.elastic.co/guide/en/enterprise-search/current/crawler-managing.html#crawler-managing-duplicate-documents)** | No           | Yes                     |
+| **Event Logging**                                                                                                                                      | No ***       | Yes                     |
+| **Monitoring**                                                                                                                                         | No           | Yes                     |
+
+*\* Open Crawler does not support [meta tags and data attributes to extract custom fields](https://www.elastic.co/guide/en/enterprise-search/current/crawler-content.html#crawler-content-meta-tags-content-extraction).*\
```
Comment (seanstory) [inline README.md:216]: Is this on the roadmap though? Is it worth signaling like you had in the blog which features are planned/scheduled even if they're not available _yet_?

File: README.md [hunk 4]
```diff
@@ -190,3 +190,33 @@ When given a command, Open Crawler will run until the process is finished.
...
+| **[Crawler directives](./docs/features/CRAWLER_DIRECTIVES.md)**                                                                                        | Yes *        | Yes                     |
+| **[Scheduling](#scheduling-recurring-crawl-jobs)**                                                                                                     | Yes          | Yes                     |
+| **Crawler results history and metadata**                                                                                                               | No **        | Yes                     |
+| **[Duplicate content handling](https://www.elastic.co/guide/en/enterprise-search/current/crawler-managing.html#crawler-managing-duplicate-documents)** | No           | Yes                     |
+| **Event Logging**                                                                                                                                      | No ***       | Yes                     |
+| **Monitoring**                                                                                                                                         | No           | Yes                     |
```
Comment (seanstory) [inline README.md:214]: This feels redundant to the `Event Logging` and `Crawl result history and metadata` rows

File: README.md [hunk 5]
```diff
@@ -190,3 +190,33 @@ When given a command, Open Crawler will run until the process is finished.
...
+
+*\* Open Crawler does not support [meta tags and data attributes to extract custom fields](https://www.elastic.co/guide/en/enterprise-search/current/crawler-content.html#crawler-content-meta-tags-content-extraction).*\
+*\*\* Crawler results history can be retained manually for Open Crawler through outputting crawl results to log files. It will not show up in Kibana.*\
+*\*\*\* Event logs can be output to the shell or a logfile for Open Crawler, but they are currently not ingested into Elasticsearch.*
+
+### Performance Comparison
+
+[See this blog post](https://www.elastic.co/search-labs/blog/elastic-open-crawler-release#how-does-the-open-crawler-compare-to-the-elastic-crawler) for a performance comparison between Open Crawler and Elastic Crawler.
```
Comment (seanstory) [inline README.md:222]: Out of curiosity, do the performance gains hold steady as you've been adding new features? I expect it's still faster, but maybe not quite as much faster?

Comment (navarone-feekery) [reply README.md:222]: Good point. I don't actually know. I'll remove this section until we have an updated performance comparison. Maybe a future blog post.

File: README.md (hunk 3 above)
Comment (navarone-feekery) [reply README.md:216]: It is on the roadmap for GA / v0.3. I'll add a mention to that (and others if they're on the roadmap).

**Comment by danajuratoni** [comment]
Could we please add more detailed extraction features to the doc e.g. URL filters, allow/disallow, CSS selectors, XPath ... (anything else added?) 
with the links to the corresponding Elastic crawler docs? 
These are recurrent questions we get once a crawler is up and running
Thanks!

**Comment by leemthompo** [comment]
🤔 Might be good to break this out onto its own page? Risks information overload on one page, plus one day we won't need this info and can just remove the page.

**Comment by navarone-feekery** [comment]
@leemthompo I've moved the table to a new page

File: docs/FEATURE_COMPARISON.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+|                                                                                               | Open Crawler                                        | Elastic Crawler (cloud)                                                                                                            |
+|-----------------------------------------------------------------------------------------------|-----------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
+| Interface                                                                                     | CLI                                                 | GUI (Kibana)                                                                                                                       |
+| Hosting                                                                                       | Self-hosted                                         | Elastic Cloud or Self-hosted Enterprise Search                                                                                     |
```
Comment (marciw) [inline docs/FEATURE_COMPARISON.md:9]:
```suggestion
| Hosting                                                                                       | Self-hosted                                         | Elastic Cloud or self-hosted Enterprise Search                                                                                     |
```

**Review by marciw — APPROVED** [review]
LGTM! Minor tweaks for clarity.

File: docs/FEATURE_COMPARISON.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+| Purge crawls                                                                                  | [Yes](../README.md#execution-logic)                 | Yes                                                                                                                                |
+| Ingest pipelines                                                                              | [Yes](./features/INGEST_PIPELINES.md)               | [Yes](https://www.elastic.co/guide/en/enterprise-search/current/crawler-managing.html#crawler-managing-ingest-pipelines)           |
+| Binary content extraction                                                                     | [Yes](./features/BINARY_CONTENT_EXTRACTION.md)      | [Yes](https://www.elastic.co/guide/en/enterprise-search/current/crawler-managing.html#crawler-managing-binary-content)             |
+| Crawl rules - allow/disallow specified URLs                                                   | [Yes](./features/CRAWL_RULES.md)                    | [Yes](https://www.elastic.co/guide/en/enterprise-search/current/crawler-managing.html#crawler-managing-crawl-rules)                |
```
[the remark this answers is not in the document]
Comment (navarone-feekery) [reply docs/FEATURE_COMPARISON.md]: I like em dashes so went with that :)

**Comment by navarone-feekery** [comment]
@seanstory @danajuratoni @marciw thanks for the reviews! I've addressed all the comments, so this is ready for re-review.

**Review by marciw — APPROVED** [review]

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**